delivery_price_product_domain/models/delivery_carrier.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2023 Cetmix OÜ - Andrey Solodovnikov
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import ast
+
+from odoo import fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class DeliveryPriceRule(models.Model):
+    _inherit = "delivery.carrier"
+
+    order_id = fields.Many2one("sale.order", string="Sale Order")
+
+    def recompute_price_available(self, apply_product, price_dict, untaxed_in_dict):
+        """This method recompute total parameters only with apply products"""
+        self.ensure_one()
+        self = self.sudo()
```

Comment:
You can't shadow existing variable.

```suggestion
        self_sudo = self.sudo()
```

And the subsequent ones

Reply:
@solo4games squash it please

Reply:
Done

Reply:
@rousseldenis go? :)